presto-main/src/main/java/com/facebook/presto/operator/scalar/IpPrefixFunctions.java

```diff
@@ -137,6 +144,222 @@ public static boolean isPrefixSubnetOf(@SqlType(StandardTypes.IPPREFIX) Slice fi
+    {
+        int inputPrefixCount = unsortedIpPrefixArray.getPositionCount();
+
+        // If we get an empty array or an array non-null single element, just return the original array.
+        if (inputPrefixCount == 0 || (inputPrefixCount == 1 && !unsortedIpPrefixArray.isNull(0))) {
+            return unsortedIpPrefixArray;
+        }
+
+        // Sort prefixes. lessThanFunction is never used. NULLs are placed at the end.
+        // Prefixes are ordered by first IP and then prefix length.
+        // Example:
+        //  Input:  10.0.0.0/8, 9.255.255.0/24, 10.0.0.0/7, 10.1.0.0/24, 10.10.0.0/16
+        //  Output: 9.255.255.0/24, 10.0.0.0/7, 10.0.0.0/8, 10.1.0.0/24, 10.10.0.0/16
+        Block ipPrefixArray = sort(null, IPPREFIX, unsortedIpPrefixArray);
+
+        // throw if anything is null
+        if (ipPrefixArray.isNull(0) || ipPrefixArray.isNull(inputPrefixCount - 1)) {
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "ip_prefix_agg does not support null elements");
+        }
+
+        // check the first and last prefixes in the array to make sure their IP versions match.
+        Slice firstIpPrefix = IPPREFIX.getSlice(ipPrefixArray, 0);
+        boolean v4 = isIpv4(firstIpPrefix);
+        Slice lastIpPrefix = IPPREFIX.getSlice(ipPrefixArray, inputPrefixCount - 1);
+        if (isIpv4(lastIpPrefix) != v4) {
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "All IPPREFIX elements must be the same IP version.");
+        }
+
+        List<List<Slice>> outputIpPrefixes = new ArrayList<>();
+        int outputPrefixCount = 0;
+        int ipMaxBitLength = v4 ? 32 : 128;
+
+        List<BigInteger[]> mergedIpRanges = mergeIpRanges(ipPrefixArray);
+        for (BigInteger[] ipRange : mergedIpRanges) {
+            List<Slice> ipPrefixes = generateMinIpPrefixes(ipRange[0], ipRange[1], ipMaxBitLength);
+            outputIpPrefixes.add(ipPrefixes);
+            outputPrefixCount += ipPrefixes.size();
+        }
+
+        BlockBuilder blockBuilder = IPPREFIX.createBlockBuilder(null, outputPrefixCount);
+        for (List<Slice> ipPrefixSlices : outputIpPrefixes) {
+            for (Slice ipPrefix : ipPrefixSlices) {
+                IPPREFIX.writeSlice(blockBuilder, ipPrefix);
+            }
+        }
+
+        return blockBuilder.build();
+    }
+
+    private static List<Slice> generateMinIpPrefixes(BigInteger firstIpAddress, BigInteger lastIpAddress, int ipVersionMaxBits)
+    {
+        List<Slice> ipPrefixSlices = new ArrayList<>();
+
+        // i.e., while firstIpAddress <= lastIpAddress
+        while (firstIpAddress.compareTo(lastIpAddress) <= 0) {
+            long rangeBits = findRangeBits(firstIpAddress, lastIpAddress); // find the number of bits for the next prefix in the range
+            int prefixLength = (int) (ipVersionMaxBits - rangeBits);
+
+            InetAddress asInetAddress;
+            try {
```

Comment:
use a longer try block to avoid splitting declaration and assignment. All code that depends on asInetAddress being set can go into the try block.